app/Services/Backups/InitiateBackupService.php

```diff
@@ -111,6 +115,17 @@ public function handle(Server $server, string $name = null): Backup
             );
         }
 
+        if (! $server->backup_limit || $server->backups()->where('is_successful', true)->count() >= $server->backup_limit) {
```

Comment:
Might want to add a comment above this line saying `"Check if the server has reached or exceeded it's backup limit"`